### DISCUSSION TITLE
Victron Smartshunt & JBD BMS as usable SOC meters

### DISCUSSION THREAD

**Discussion by intothewildmike** [Louisvdw/dbus-serialbattery#662] [Q&A]
I am busy with a Portable 280AH 4s LiFep04 battery DIY build that will include a JBD BMS, External Heltec active balancer and a Victron SmartShunt (which I have yet to purchase) all attached to one unit.

I want to use the battery in my camper with an all Victron setup - Multiplus, Orion, Smart MPPT and Cerbo GX.  When using the battery in my camper does it make sense to use the JBD BMS as a shunt to control my system via the Cerbo RS485 port? How would this work if I also had a Victron Smart shunt installed and attached to the Cerbo ? Would things clash? Should I just leave the Smartshunt unconnected to the Cerbo ?

The reason I was thinking to install the SmartShunt in any case, is when the camper is stored I will use the battery unit in my house connected to a Xantrax Inverter (which I have) and victron charger to use during load-shedding and I would like to accurately monitor the SOC. So in this case I could monitor the SOC status via the SmartShunt.  

Would this scenario I am attempting to describe above work? Is there a better way to monitor the SOC of the battery when it is not plugged into the all Victron camper ecosystem?

Any suggestions, tips or advice would be appreciated.

Michael

**Comment by dfch** [comment]
Hi @intothewildmike trying to answer your questions:

> When using the battery in my camper does it make sense to use the JBD BMS as a shunt to control my system via the Cerbo RS485 port?

If you are only using a single battery and you *trust* the JBD to correctly measure the current, then you could use only the JBD. However, if you are using additional DC loads on your battery, you might have to make additional configuration in the GX so reflect that.

> How would this work if I also had a Victron Smart shunt installed and attached to the Cerbo ? 

In Venus OS as far as I know only a single battery monitor can be active/selected. So even when you have a SmartShunt *and* a JBD as a monitor you will have to decide which one to use. 

> Would things clash? 

I do not think things would clash. You would only be able to use either or as a source of information. The JBD should be able to give you more detailed information as the SmartShunt does not have insight into the actually battery (but only measures the current). And you would potentionally see two devices giving you different information regarding SoC

> Should I just leave the Smartshunt unconnected to the Cerbo ?

If you leave the SmartShunt unconnected I would rather not install it at all. This would result in a setup with reduced complexity.

> Would this scenario I am attempting to describe above work?

In my opinion it should work, but I would probably not do it that way (see below for further explanation).

> Is there a better way to monitor the SOC of the battery when it is not plugged into the all Victron camper ecosystem?

You can still use just the JDB app to *observe* the SOC (if this is what you mean by *monitor*).

Personally, I think the idea of moving the battery with the SmartShunt to the house to connect to a different inverter *and* charger whereas the MultiPlus is an inverter/charger (in a single device) seems a little bit complicated to me. In addition, the SmartShunt might lose its SoC info when moved around and powered off (see [10.3.5. State of charge is missing](https://www.victronenergy.com/media/pg/SmartShunt/en/troubleshooting.html#UUID-f579e0ce-9f28-44cb-11e4-2de07d261043) ) and the JBD would be running all the time and therefore would keep the SoC information. 
So, I would rather 
* only move the battery and not have a SmartShunt at all;
  - easiest in regards to re-cabling and lifting, moving
*  or move the battery along with the Cerbo;
  - with that you would have the same usability regardless of the place of its operation. Plus with the GX (as you were writing you would be using a Victron charger for your battery in your house), you could still use DVCC if that charger has a VE.Bus
  - requires a more advanced charger
  - more in regards to re-cabling and lifting, moving
* or move the MultiPlus along with the Battery.
  - which would save you two additional devices you mentioned (and even if you already have them, you could use them elsewhere
  - same usability regardless of location
  - same maintainability
  - even more in regards to re-cabling and lifting, moving
  - use DVCC for charging and discharging

This is just my opinion based on the information you provided. There might be more to it or your use case might differ. Hope this helps anyway.

ps I think your questions would rather fit better in some kind of community forum. They are only very vaguely connected to the dbus-serialbattery driver and more regarding the design of a system in general.

**Comment by spiralj** [comment]
Hi @intothewildmike  -
I'm curious what you ended up doing. I also wanted to have the Victron Smart Shunt be the source of SOC in Cerbo, but wanted to be able to see the additional data from the JBD in VenusOS.
Are you discussing this anywhere else as @dfch suggested?

**Reply by TheGrumpyBadger** [reply]
+1 for this, I would like zu use die JKBMS to monitor the cells but the SOC is wrong after some days without 100% SOC.
It would be great to overwrite the SoC from the JK BMS with the one from a SmartShunt.

**Reply by mr-manuel** [reply]
You can use [dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery) to create a virtual battery and combine the data you want on your own. To achieve this you can use Node-RED which is part of Venus OS Large. Just Google for it :)

**Comment by squowse** [comment]
> You can use [dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery) to create a virtual battery and combine the data you want on your own. To achieve this you can use Node-RED which is part of Venus OS Large. Just Google for it :)

@mr-manuel 

Thanks - is this still the best way to do this? I wish to use the smartshunt for SoC for greater accuracy. Do the BMS and Smartshunt need to be publishing information on MQTT first? Or will the MQTT battery ssort that out? Sorry I don't know much about this process.

**Reply by mr-manuel** [reply]
With the latest nightly in my repo you can use the current measurement of the SmartShunt and the SOC calculation of the driver. Read the changelog and `config.default.ini` for more info.

https://github.com/mr-manuel/venus-os_dbus-serialbattery

**Reply by squowse** [reply]
very interesting. So the serial battery driver would do the SoC calculations? It would continuously use the smartshunt current values to do couloumb counting? Would this be easier than using the smartshunt's built in SoC calculation?

**Reply by mr-manuel** [reply]
At the moment this would be easier

**Reply by squowse** [reply]
Thanks. I’m going to try this. One question- what do I tell VRM that the Smartshunt is? If I don’t allocate it as battery monitor will it just ignore it?

**Reply by mr-manuel** [reply]
The VRM displays only the selected battery monitor in the overview.

**Comment by intothewildmike** [comment]
Hello Guys, I never got round to building the battery or system as described above however I did however pick up a very decent battery for a good price brand new from Auction. Battery is rebranded version of the Poweroad PRLC-100. 
<img width="564" alt="Screenshot 2024-11-19 at 12 14 13" src="https://github.com/user-attachments/assets/ceef8982-ea0e-4ed4-b3c7-b0cc3806f04b">

I am trying to figure out how best to have the battery BMS communicate and show an accurate SOC reading in the VRM (which i would assume will be more accurate and thus negate the need for an external shunt such as BMV712) with my Cerbo GX and rest of my devices (Multiplus, solar charger 100-50 and DC-DC orion XS) 

Here are pictures and screenshots the manufacturer has sent to me which I am trying to make sense of:

![WhatsApp Image 2024-11-19 at 10 41 51](https://github.com/user-attachments/assets/20149298-e92e-402d-8a5b-f0fd5de036d2)
![WhatsApp Image 2024-11-19 at 10 42 12](https://github.com/user-attachments/assets/11f99a7c-2a27-43e0-a96b-cd4025a14c84)
![WhatsApp Image 2024-11-19 at 10 51 51](https://github.com/user-attachments/assets/2179daec-bc08-4986-838f-6e6c66959a1b)
![WhatsApp Image 2024-11-19 at 10 59 55](https://github.com/user-attachments/assets/40696e84-2ea7-4b39-b7cd-1a579fe2c0f2)
![WhatsApp Image 2024-11-19 at 10 59 27](https://github.com/user-attachments/assets/100c845c-815f-4392-88c0-3748f8ca7c33)
![WhatsApp Image 2024-11-19 at 11 19 38](https://github.com/user-attachments/assets/1eda88b2-1632-4850-8352-1993b47b0f99)
![WhatsApp Image 2024-11-19 at 11 28 52](https://github.com/user-attachments/assets/9d642fd5-53d4-4792-99b0-594c13fa2290)
![WhatsApp Image 2024-11-19 at 11 02 41](https://github.com/user-attachments/assets/80e161df-d1ba-438c-bf69-8ac5df110ac8)

They saying I have to make a custom cable "Wiring harness to be customized. One side  RJ45  another side RJ11"

Can anyone help me make sense of what I need to do here? Would be grateful.

Michael

**Comment by spiralj** [comment]
It's my understanding you still need a smart shunt.

**Comment by intothewildmike** [comment]
Have spoken to the manufacturer of the battery who also make the BMS.  I am told I need to use a Canbus tool to update the firmware of the battery BMS and then it will be picked and communicates on the Cerbo GX. 
<img width="360" alt="Screenshot 2024-11-25 at 10 43 25" src="https://github.com/user-attachments/assets/0d7cda9d-ca2c-494b-9d96-b27aa5619552">
<img width="424" alt="Screenshot 2024-11-25 at 10 43 36" src="https://github.com/user-attachments/assets/6f0b0d94-faa4-47ec-a67e-86556e2a0820">
<img width="435" alt="Screenshot 2024-11-25 at 10 43 42" src="https://github.com/user-attachments/assets/9c0be1f9-8ee1-440d-8ba9-d3fc4ea43e64">

Does anyone in Cape Town know where I can get one of these tools - Happy also to rent one to update the firmware and return it.
@Louisvdw perhaps you may know someone in South Africa - I am based in Cape Town

**Reply by mr-manuel** [reply]
You could buy one on [Aliexpress](https://www.aliexpress.com/w/wholesale-USB-to-CAN.html) for about 10 USD.

**Reply by Louisvdw** [reply]
You can get this one from [Micro Robotics](https://www.robotics.org.za/MKS-CAN-V2-STD?tracking=MrYShlotRq43KUyHzpiDk7TuNTqqwUOQvDY6gvcl0ga7UzTx4uz5Zouk4Akylf4v) there in Stellenbosch. It should work the same.
